New printf warnings in RTEMS build

**Describe the bug**
A recent change re-introduced some printf type warnings on the 32-bit RTEMS build

**To Reproduce**
Build for RTEMS i686 target
Warnings are issued:
```
[ 65%] Building C object cfe-core/CMakeFiles/cfe-core.dir/src/sb/cfe_sb_api.c.o
/home/joe/code/cfecfs/github/cfe/fsw/cfe-core/src/sb/cfe_sb_api.c: In function 'CFE_SB_SubscribeFull':
/home/joe/code/cfecfs/github/cfe/fsw/cfe-core/src/sb/cfe_sb_api.c:899:9: error: format '%u' expects argument of type 'unsigned int', but argument 6 has type 'uint32' [-Werror=format=]
         (unsigned int)CFE_SB_MsgIdToValue(MsgId), Collisions);
         ^
cc1: all warnings being treated as errors
```

**Expected behavior**
Should build cleanly

**System observed on:**
Ubuntu 20.04 build host for i686-rtems4.11 target.

**Additional context**
This always happens with fixed width types, needs explicit cast when used with printf.

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

--- a/fsw/cfe-core/src/sb/cfe_sb_api.c
+++ b/fsw/cfe-core/src/sb/cfe_sb_api.c
@@ -896,7 +896,7 @@ int32  CFE_SB_SubscribeFull(CFE_SB_MsgId_t   MsgId,
     {
         CFE_EVS_SendEventWithAppID(CFE_SB_HASHCOLLISION_EID, CFE_EVS_EventType_DEBUG, CFE_SB.AppId,
         "Msg hash collision: MsgId = 0x%x, collisions = %u",
-        (unsigned int)CFE_SB_MsgIdToValue(MsgId), Collisions);
+        (unsigned int)CFE_SB_MsgIdToValue(MsgId), (unsigned int)Collisions);
     }
 
     return CFE_SUCCESS;

--- a/modules/sbr/unit-test-coverage/test_cfe_sbr_map_direct.c
+++ b/modules/sbr/unit-test-coverage/test_cfe_sbr_map_direct.c
@@ -99,7 +99,7 @@ void Test_SBR_Map_Direct(void)
             count++;
         }
     }
-    UtPrintf("Valid route id's encountered in performance loop: %u", count);
+    UtPrintf("Valid route id's encountered in performance loop: %u", (unsigned int)count);
 }
 
 /* Main unit test routine */

--- a/modules/sbr/unit-test-coverage/test_cfe_sbr_map_hash.c
+++ b/modules/sbr/unit-test-coverage/test_cfe_sbr_map_hash.c
@@ -105,7 +105,7 @@ void Test_SBR_Map_Hash(void)
             count++;
         }
     }
-    UtPrintf("Valid route id's encountered in performance loop: %u", count);
+    UtPrintf("Valid route id's encountered in performance loop: %u", (unsigned int)count);
 }
 
 /* Main unit test routine */

